src/data/badge.ts

```diff
@@ -0,0 +1,39 @@
+import type { BadgeGrade } from "#enums/badge-grades";
+import type { BadgeType } from "#enums/badge-type";
+import type { Type } from "#enums/type";
+import i18next from "i18next";
+
+/**
+ * Badges are key items that are given to the player and affect the run
+ *
+ * @param id the corresponding BadgeType enum
+ * @param localeKey the locale key of the badge
+ * @param iconImage the string representation of the image
+ * @param generation which generation the badge originated
+ * @param type the pokemon type associated with the badge
+ * @param grade a way to tweak how effective a badge is
+ */
+export class Badge {
+  public id: BadgeType;
+  public localeKey: string;
+  public iconImage: string;
+  public generation: number;
+  public type: Type;
+  public grade: BadgeGrade;
+
+  constructor(id: BadgeType, localeKey: string, iconImage: string, generation: number, type: Type) {
+    this.id = id;
+    this.localeKey = localeKey;
+    this.iconImage = iconImage;
+    this.generation = generation;
+    this.type = type;
+  }
+
+  getName(): string {
```

Comment:
```suggestion
  public get name(): string {
```

i.e. you can use `Badge.name` to get a localized name for the badge.

Reply:
the `getThing()` and `get thing()` discrepancy is a thing in all of the codebase. We should decide on which pattern to use. 

re: `trainer` linking, yeah we can definitely have that on the Badge class. Not like extra information will hurt

Reply:
I find `get thing()` to be cleaner personally, but `getThing()` would still be needed in cases where parameters are needed. `get thing(foo)` doesn't compile afaik.